src/main/java/core/basesyntax/AbstractFigure.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax;
+
+public abstract class AbstractFigure implements Figure {
+    private static final String FIGURE_INFO_FORMAT = "Figure: %s, area: %.2f sq. units, color: %s";
+    private final String color;
+
+    public AbstractFigure(String color) {
+        this.color = color;
+    }
+
+    @Override
+    public abstract double getArea();
+
+    @Override
+    public void draw() {
+        String figureInfo = String.format(FIGURE_INFO_FORMAT,
+                this.getClass().getSimpleName(), getArea(), color);
+        System.out.println(figureInfo);
```

Comment:
You can print it immediately, without creating this temporary `figureInfo` variable